ruby/ext/google/protobuf_c/map.c

```diff
@@ -71,6 +71,9 @@ static VALUE table_key(Map* self, VALUE key,
     case UPB_TYPE_BYTES:
     case UPB_TYPE_STRING:
       // Strings: use string content directly.
+      if (TYPE(key) == T_SYMBOL) {
```

Comment:
Why we have to this in order to support Struct?

Reply:
I don't think so, and it didn't break any tests that I'm aware of. Can you think of a case where it would make a difference?

This would not throw type errors in some cases where it would have previously but I don't know if that should be considered additive vs a breaking change?

Reply:
It doesn't break existing test cases. However, it allows symbol for string map, which is not allowed previously.

Reply:
Right, is there any reason to prohibit that?

Reply:
It makes map not consistent with optional, repeated fields.
Besides, how to compare symbol? identity compare? value compare? Is the semantic same as string?
Besides, is the feature quite important? Otherwise, we don't want to over-provide it, as these trivial features necessarily makes our library complicate.

Reply:
It also breaks existing type checking. We should added tests for our type checking.

Reply:
It allows users to provide:

```ruby
Google::Protobuf::Struct.new(fields: {
   a: {number_value: 2.0}
})
```

instead of:

```ruby
Google::Protobuf::Struct.new(fields: {
   'a' => {number_value: 2.0}
})
```

I would argue that Converting `Symbol` key values to `String` in `Struct` is a beneficial affordance, as `Struct` will only every have `String` keys. `Map` can have any any value type as a key, so it wouldn't make as much sense.

That said, I do think it would be fine to accept `Symbol` values every time a `String` is expected. In Ruby, I would write it like this:

```ruby
class Google::Protobuf::Struct
  def [](key)
    inner_struct[String(key)]
  end
end
```

Basically, anytime we expect a `String` object, pass it to the `String()` method to perform the necessary conversions.